org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java

```diff
@@ -260,6 +278,9 @@ public void launch(DSPLaunchDelegateLaunchBuilder builder) throws CoreException
 				}
 
 				ProcessBuilder processBuilder = new ProcessBuilder(command);
+				if (builder.debugCmdEnvVars != null && !builder.debugCmdEnvVars.isEmpty()) {
```

Comment:
PS. The alternative to reformatting the environment array is to change the launch itself to use `DebugPlugin.exec` instead which expects that format. The DebugPlugin.exec just uses ProcessBuilder anyway.

Reply:
Good idea. thanks!